app.go

```diff
@@ -328,19 +333,9 @@ func (a *App) RunAndExitOnError() {
 // RunAsSubcommand invokes the subcommand given the context, parses ctx.Args() to
 // generate command-specific flags
 func (a *App) RunAsSubcommand(ctx *Context) (err error) {
+	// Setup also handles HideHelp and HideHelpCommand
 	a.Setup()
 
-	// append help to commands
-	if len(a.Commands) > 0 {
-		if a.Command(helpCommand.Name) == nil && !a.HideHelp {
-			a.appendCommand(helpCommand)
-
-			if HelpFlag != nil {
-				a.appendFlag(HelpFlag)
-			}
-		}
-	}
```

Comment:
❓ _(blocking question)_ this code is being removed because it was duplicated in `Setup` above, yeah?

Reply:
Yes, and because of the duplication, this code path was not executed and could not be covered with tests: https://github.com/urfave/cli/pull/1083#discussion_r388284552

Reply:
Nice find!